src/components/layout/NavBar/index.tsx

```diff
@@ -0,0 +1,112 @@
+import Assignment from '@/components/@common/SVG/Icon/Assignment';
+import Home from '@/components/@common/SVG/Icon/Home';
+import Logo from '@/components/@common/SVG/Icon/Logo';
+import Profile from '@/components/@common/SVG/Icon/Profile';
+import Search from '@/components/@common/SVG/Icon/Search';
+import Setting from '@/components/@common/SVG/Icon/Setting';
+import {
+  BOARD_PATH,
+  ROOT_PATH,
+  SEARCH_PATH,
+  SETTING_PATH,
+  USER_PATH,
+} from '@/constants/routes';
+import { Link } from 'react-router';
+import { v4 as uuidv4 } from 'uuid';
+import * as Styled from './NavBar.styled';
+import Collapse from '@/components/@common/SVG/Icon/Collapse';
+import HomeFill from '@/components/@common/SVG/Icon/HomeFill';
+import AssignmentFill from '@/components/@common/SVG/Icon/AssignmentFill';
+import ProfileFill from '@/components/@common/SVG/Icon/ProfileFill';
+import SettingFill from '@/components/@common/SVG/Icon/SettingFill';
+import { useRecoilState } from 'recoil';
+import isSideBarOpenAtom from '@/recoil/layout/isSideBarOpen';
+import Expand from '@/components/@common/SVG/Icon/Expand';
+import SearchFill from '@/components/@common/SVG/Icon/SearchFill';
+
+const navMenus = [
+  {
+    Icon: Home,
+    ActiveIcon: HomeFill,
+    title: '홈',
+    path: ROOT_PATH.ROOT,
+  },
+  {
+    Icon: Search,
+    ActiveIcon: SearchFill,
+    title: '검색',
+    path: SEARCH_PATH.ROOT,
+  },
+  {
+    Icon: Assignment,
+    ActiveIcon: AssignmentFill,
+    title: '게시물',
+    path: BOARD_PATH.ROOT,
+  },
+  {
+    Icon: Profile,
+    ActiveIcon: ProfileFill,
+    title: '마이페이지',
+    path: USER_PATH.PROFILE,
+  },
+  {
+    Icon: Setting,
+    ActiveIcon: SettingFill,
+    title: '설정',
+    path: SETTING_PATH.ROOT,
+  },
+];
+
+const NavBar = () => {
+  const [isSideBarOpen, setIsSideBarOpen] = useRecoilState(isSideBarOpenAtom);
+
+  const handleSideBarToggle = () => {
+    setIsSideBarOpen((prev) => !prev);
+  };
+  return (
+    <Styled.Wrapper>
+      <Link to={ROOT_PATH.ROOT}>
+        <Logo title="logo" width="48px" height="48px" />
+      </Link>
+      <Styled.NavMenuList>
+        {navMenus.map(({ Icon, ActiveIcon, title, path }) => (
+          <Styled.NavMenu key={uuidv4()}>
```

Comment:
```suggestion
const id = useId();
...
return(
    ...
  <Styled.NavMenu key={id}>
)     
```
uuid 대신 **useId** 훅을 사용해도 좋을 것 같습니다!

Reply:
`useId`를 사용해서 자식 컴포넌트에 전달하면 `<NavBar>`에서 생성된 id값 하나가 자식 컴포넌트들에게 전달되는 것으로 알고있습니다.

공식문서에서도 프론트에서 만든 데이터의 경우 라이브러리를 사용하는 것을 추천하고 있는것 같아요. 아래 링크에 들어가시면 바로 확인하실 수 있습니다!!
<img width="922" alt="스크린샷 2025-01-08 오전 9 23 14" src="https://github.com/user-attachments/assets/050ee45c-fe55-44e3-8a5a-d0bb33b10399" />

https://ko.react.dev/learn/rendering-lists#where-to-get-your-key

Reply:
오호 저도 지원님 피드백을 보고 `useId` 훅과 `uuid`는 어떤 차이가 있나 봤더니
공식 문서에서 `useId`는 리스트 내에서 key 생성 목적의 사용은 지양하고 있네요!

이것도 하나 배워갑니닷 🙇

Reply:
useId는 하나의 컴포넌트의 모든 렌더링에서는 동일한 id를 반환하는군요..!
덕분에 하나 배웠습니다👍👍